### PR TITLE
Added required models and Login / signup feature in query and mutation

### PR DESCRIPTION
**Pull request by bhumisha**

### CODE REVIEW

File: server/utils/auth.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+const jwt = require('jsonwebtoken');
+
+const secret = 'mysecretsTokenStringss';
```
Comment (yulduzetta) [inline server/utils/auth.js:3]: could we add this to .env file instead? I'll make sure to also add it to prod env vars set on heroku

File: server/schemas/typeDefs.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  phone:Int
+}
+
+type ParkingPlaces {
+  _id: ID
+  apt: String
+  street: String
+  city: String
+  state: String
+  zip:String
```
Comment (yulduzetta) [inline server/schemas/typeDefs.js:27]: could we externalize these properties into Address schema? We'd need it later for processing stripe transaction for a user

**Comment by bhumisha** [comment]
Added AddressSchema and moved variable to .env
defined add_ParkingPlace mutation and resolver is inprogress so logic commented

**Comment by bhumisha** [comment]
mutation addUser($firstName: String!,$lastName: String!, $email: String!, $password: String!,$phone:String!) {
    addUser(firstName : $firstName, lastName :$lastName, email: $email, password: $password, phone:$phone) {
      token
      user {
        _id
        firstName  lastName
        email
      }
    }
  }
**Query Variable**
{
"firstName":"test",
  "lastName":"dave",
  "email":"test@gmail.com",
  "password":"test123",
  "phone":"404-444-4433"
}


mutation login($email: String!, $password: String!) {
  login(email: $email, password: $password) {
    token
    user {
      _id
    }
  }
}
Query Variable
{
  "email":"test@gmail.com",
  "password":"test123"
}
{
  "authorization" : "bearer Add your token from previous mutation"
}

**Review by yulduzetta — APPROVED** [review]
gtg! as discussed, we'll adjust if needed!

**Review by akramsabbah9 — APPROVED** [review]
Looks good to me!

**Comment by yulduzetta** [comment]
> mutation addUser($firstName: String!,$lastName: String!, $email: String!, $password: String!,$phone:String!) {
> addUser(firstName : $firstName, lastName :$lastName, email: $email, password: $password, phone:$phone) {
> token
> user {
> _id
> firstName lastName
> email
> }
> }
> }
> **Query Variable**
> {
> "firstName":"test",
> "lastName":"dave",
> "email":"[test@gmail.com](mailto:test@gmail.com)",
> "password":"test123",
> "phone":"404-444-4433"
> }
> 
> mutation login($email: String!, $password: String!) {
> login(email: $email, password: $password) {
> token
> user {
> _id
> }
> }
> }
> Query Variable
> {
> "email":"[test@gmail.com](mailto:test@gmail.com)",
> "password":"test123"
> }
> {
> "authorization" : "bearer Add your token from previous mutation"
> }

thanks for sharing. For login mutation tho, we don't need to provide the auth token, as the login operation returns the new token each time:
```
login: async (parent, { email, password }) => {
      const user = await User.findOne({ email });
      if (!user) {
        throw new AuthenticationError("Incorrect credentials");
      }
      const correctPw = await user.isCorrectPassword(password);

      if (!correctPw) {
        throw new AuthenticationError("Incorrect credentials");
      }

      const token = signToken(user);

      return { token, user };
    },
```

**Pull request merged**